### ISSUE TITLE
bug in kalman smoother when k=0?

### ISSUE THREAD

**Issue by rBatt** [GLEON/LakeMetabolizer#92]
https://github.com/GLEON/LakeMetabolizer/blob/master/R/metab.kalman.R#L295

[The commit (ryan's) that created the bug is here](https://github.com/GLEON/LakeMetabolizer/commit/a22c05eec2ac95ae480dbcbfae82ba1226de5d68#diff-04784787209a24bc5bb013217ccc0da6R163)

Resulted from trying to handle k=0. That's a bug, right??

@lawinslow 

@cbuelo You said you'd had an issue with smoothing the oxygen, right? It might be that the earlier versions I tested hadn't yet tried to handle k=0, I'd have to check.

Note that this only affects the smoother of DO, not the estimation of metabolism.

**Comment by cbuelo** [comment]
Yeah if I remember correctly the returned "smooth" DO time series had big, random-looking jumps in it.

**Comment by rBatt** [comment]
That's probably due to dividing by a small fraction in the smoother, but
I'm not sure.

These issues may be unrelated. However, I'd be interested to see if your
smoothing problem was solved once this bug fix is implemented.

**Comment by rBatt** [comment]
This is now related to #126
